benchmarks/distributed/rpc/parameter_server/launcher.py

```diff
@@ -185,22 +202,23 @@ def run_benchmark(rank, model, data, config):
         )
     else:
         # trainers = [0, trainer_count)
-        trainer_config = config.trainer_config
-        ps_config = config.ps_config
-        if (USE_CUDA_RPC in trainer_config and
-            trainer_config[USE_CUDA_RPC] and
-            USE_CUDA_RPC in ps_config and
-            ps_config[USE_CUDA_RPC] and
-                config.ps_count > 0):
-            ps_rank = get_parameter_server_rank(rank, config)
-            ps_name = get_name(
-                ps_rank,
-                config
-            )
-            rpc_backend_options.set_device_map(
-                ps_name,
-                {rank: ps_rank}
-            )
+        trainer_configs = config.trainer_config["configurations"]
+        if config.ps_count > 0:
+            ps_configs = config.ps_config["configurations"]
+            if (USE_CUDA_RPC in trainer_configs and
+                trainer_configs[USE_CUDA_RPC] and
+                USE_CUDA_RPC in ps_configs and
```

Comment:
what if only one of trainer and ps wants to use CUDA RPC?

Reply:
The benchmark was designed so that all the servers and trainers will either use CUDA RPC or CPU RPC. 

I'll add benchmark arguments for the number of servers and trainers that use CUDA RPC.

Reply:
I see, this means we should error out when there are conflicts?

Reply:
I have added assertions on trainer and server counts.

I haven't been able to test using a mixed configuration. For example 1 cputrainer 1 cpuserver and 1 cudatrainer 1 cudatrainer because of CUDA RPC not implemented error.